doc/docs/master/deploy-manage/deploy/envoy.md

```diff
@@ -0,0 +1,487 @@
+
+!!! Note 
+    See our [reference values.yaml](https://github.com/pachyderm/pachyderm/blob/master/etc/helm/pachyderm/values.yaml#L649){target=_blank} for all available configurable fields of the proxy.
+
+Before any deployment in production, we recommend reading the following recommendations section and [set up your production infrastructure](#deploy-pachyderm-in-production-with-envoy). 
+
+Alternatively, you can skip those infrastructure prerequisites and make a [quick cloud installation](#quick-cloud-deployment-with-envoy) or jump to our [local deployment](#deploy-pachyderm-locally-with-envoy) section for the first encounter with Pachyderm.
+
+## Pachyderm General Infrastructure Recommendations
+
+For production deployments, we recommend that you:
+
+* **Provision a TCP load balancer** for all HTTP/HTTPS, gRPC/gRPCs, aws s3, /dex incoming traffic.
+The TCP load balancer (load balanced at L4 of the OSI model) will have port `80/443` forwarding to the `pachyderm-proxy` service entry point. Please take a look at the diagram above.
+        
+    When Envoy is enabled with `type:LoadBalancer` (see the snippet of values.yaml enabling the proxy), Pachyderm creates a `pachyderm-proxy` service allowing your cloud platform (AWS, GKE...) to **provision a TCP Load Balancer automatically**.
+        
+    To provision this external load balancer automatically (if supported) and attach any Load Balancer configuration information to the metadata of your servic, add the appropriate `annotations` in the `proxy.service` of your values.yaml (see below):
+
+    ```yaml
+    proxy:
+      enabled: true
+      service:
+        type: LoadBalancer
+        annotations: {see examples below}
+    ```
+
+
+    === "Example on AWS EKS"
+        In the following example, we deploy an NLB on AWS EKS:
+
+        ``` yaml
+        proxy:
+          enabled: true
+          service:
+            type: LoadBalancer
+            annotations: {see examples below}
+              service.beta.kubernetes.io/aws-load-balancer-type: "external"
+              service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "ip"
+              service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
+              service.beta.kubernetes.io/aws-load-balancer-subnets: "subnet-aaaaa,subnet-bbbbb,subnet-ccccc"
+
+        ```
+
+    === "Example on GCP GKE"
+        In the following example, we pre-created a static IP by running `gcloud compute addresses create ADDRESS_NAME --global --ip-version IPV4`, then passed this external IP to the values.yaml as follow:
+
+        ``` yaml
+        proxy:
+          enabled: true
+          service:
+            type: LoadBalancer
+            loadBalancerIP: ${ADDRESS_NAME}
```

Comment:
Does this name based binding work?  I assume you have to do something like:

```
$ gcloud compute addresses create jonathan-test-address-deleteme
Created [https://www.googleapis.com/compute/v1/projects/kyle-dev-237522/regions/us-central1/addresses/jonathan-test-address-deleteme].

$ gcloud compute addresses describe jonathan-test-address-deleteme
address: 34.67.7.54
addressType: EXTERNAL
creationTimestamp: '2022-06-03T14:07:02.870-07:00'
description: ''
id: '3059108090574522649'
kind: compute#address
name: jonathan-test-address-deleteme
networkTier: PREMIUM
region: https://www.googleapis.com/compute/v1/projects/kyle-dev-237522/regions/us-central1
selfLink: https://www.googleapis.com/compute/v1/projects/kyle-dev-237522/regions/us-central1/addresses/jonathan-test-address-deleteme
status: RESERVED
```

And then put `34.67.7.54` in the loadBalancerIP field.